keps/sig-azure/0024-20180809-cross-resource-group-nodes.md

```diff
@@ -0,0 +1,165 @@
+---
+kep-number: 24
+title: Cross resource group nodes
+authors:
+  - "@feiskyer"
+owning-sig: sig-azure
+participating-sigs:
+  - sig-azure
+reviewers:
+  - name: "@khenidak"
+  - name: "@colemickens"
+approvers:
+  - name: "@brendanburns"
+editor: TBD
+creation-date: 2018-08-09
+last-updated: 2018-08-09
+status: provisional
+---
+
+# Cross resource group nodes
+
+## Table of Contents
+
+<!-- TOC -->
+
+- [Cross resource group nodes](#cross-resource-group-nodes)
+    - [Table of Contents](#table-of-contents)
+    - [Summary](#summary)
+    - [Motivation](#motivation)
+        - [Assumptions](#assumptions)
+        - [Non-Goals](#non-goals)
+    - [Design](#design)
+    - [Implementation](#implementation)
+        - [Cross-RG nodes](#cross-rg-nodes)
+        - [On-prem nodes](#on-prem-nodes)
+    - [Alternatives](#alternatives)
+
+<!-- /TOC -->
+
+## Summary
+
+This KEP aims to add cross resource group (RG) nodes and on-prem nodes support to Azure cloud provider.
+
+## Motivation
+
+Today, Azure cloud provider only supports nodes from specified RG (which is set in cloud provider configure file). For nodes in different RG, Azure cloud provider reports `InstanceNotFound` error and thus they would be removed by controller manager. So as on-prem nodes.
+
+With managed cluster (limited access to nodes) users may need nodes where they can customize in ways that are not possible in a managed service. So this document proposes to support joining arbitrary nodes to the cluster and make required changes for both Azure cloud provider and provision setups, which includes
```

Comment:
With managed clusters, like [AKS](https://docs.microsoft.com/en-us/azure/aks/), there is limited access to configure nodes. There are instances where users may need to customize nodes in ways that are not possible in a managed service.**

Reply:
This document proposes support for joining arbitrary nodes to a cluster and the required changes to make in both the Azure cloud provider and provisioned setups, which include:**